maintainers/scripts/update.nix

```diff
@@ -132,19 +136,28 @@ let
         --argstr keep-going true
 
     to continue running when a single update fails.
+
+    You can also make the updater automatically commit on your behalf from updateScripts
+    that support it by adding
+
+        --argstr commit true
   '';
 
-  packageData = package: {
+  packageData = { package, attrPath }: {
     name = package.name;
     pname = lib.getName package;
-    updateScript = map builtins.toString (lib.toList package.updateScript);
+    oldVersion = lib.getVersion package;
+    updateScript = map builtins.toString (lib.toList (package.updateScript.command or package.updateScript));
+    supportedFeatures = package.updateScript.supportedFeatures or [];
+    attrPath = package.updateScript.attrPath or attrPath;
```

Comment:
It'd be nice if this were documented.

Reply:
I intentionally removed the documentation in https://github.com/NixOS/nixpkgs/commit/4bc26cf979e1f39bdee232a357134faa9fb8dfcf so that people do not rely on it yet. I am weighting whether the added complexity is worth it and reserve the right to remove it.